From ef0fdb83b89c87b63e94bf6ecdab5cd8c6458b2e Mon Sep 17 00:00:00 2001
From: Jay Satiro <raysatiro@yahoo.com>
Date: Thu, 18 Jun 2015 19:35:04 -0400
Subject: [PATCH] cookie: Fix bug in export if any-domain cookie is present

In 3013bb6 I had changed cookie export to ignore any-domain cookies,
however the logic I used to do so was incorrect, and would lead to a
busy loop in the case of exporting a cookie list that contained
any-domain cookies. The result of that is worse though, because in that
case the other cookies would not be written resulting in an empty file
once the application is terminated to stop the busy loop.
---
 lib/cookie.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git lib/cookie.c lib/cookie.c
index 94f2a8b..22730cf 100644
--- lib/cookie.c
+++ lib/cookie.c
@@ -1274,9 +1274,8 @@
           "# http://curl.haxx.se/docs/http-cookies.html\n"
           "# This file was generated by libcurl! Edit at your own risk.\n\n",
           out);
-    co = c->cookies;
 
-    while(co) {
+    for(co = c->cookies; co; co = co->next) {
       if(!co->domain)
         continue;
       format_ptr = get_netscape_format(co);
@@ -1288,7 +1287,6 @@
       }
       fprintf(out, "%s\n", format_ptr);
       free(format_ptr);
-      co=co->next;
     }
   }
 
@@ -1309,9 +1307,7 @@
       (data->cookies->numcookies == 0))
     return NULL;
 
-  c = data->cookies->cookies;
-
-  while(c) {
+  for(c = data->cookies->cookies; c; c = c->next) {
     if(!c->domain)
       continue;
     line = get_netscape_format(c);
@@ -1326,7 +1322,6 @@
       return NULL;
     }
     list = beg;
-    c = c->next;
   }
 
   return list;
